### DISCUSSION TITLE
The touch screen doesn't work well.

### DISCUSSION THREAD

**Discussion by js-midi** [shanteacontrols/OpenDeck#98] [Q&A]
The touch screen doesn't work well.
What's the problem?
I followed the instructions in the wiki, but there is a problem.
https://youtu.be/kaoyCZU9lOY

**Comment by paradajz** [comment]
I don't have enough information. Did you enable it in Web configurator? Which firmware are you using?

-------- Original Message --------

**Reply by js-midi** [reply]
The default firmware version was 5.55.
In the nextion editor and web configurator, I set the same as described.

1: Is there anything else you need to do?
2: Is there any part that needs to be coded separately in the firmware?
 
※ I used nextion 4.3 inch, stm32 blackfill 411, firmware 5.55

**Reply by js-midi** [reply]
https://youtu.be/CtpuxTsN-_g

**Reply by js-midi** [reply]
https://youtube.com/shorts/N5zo9q0-BSY?feature=share

**Reply by js-midi** [reply]
As a result of the test, it seems that the hotspot element coordinates are all mapped to the first icon in the web configurator of each icon.
Each coordinate value is set correctly.

**Reply by paradajz** [reply]
Please send me your touchscreen design file.

**Reply by js-midi** [reply]
Touchscreen design file.
The same thing happens with the downloaded cardamom file.
[js-midi.zip](https://github.com/shanteacontrols/OpenDeck/files/7117890/js-midi.zip)

**Reply by paradajz** [reply]
Your video shows that the touchscreen works as it should - different buttons on the screen trigger different buttons in the firmware / UI. But then you write that each button triggers the same elements? I don't understand the issue that you're having.

**Reply by paradajz** [reply]
Also you have sent me .tft file - I need HMI file.

**Reply by js-midi** [reply]
All in the zip file

**Reply by js-midi** [reply]
HMI ,tft ,sysex

**Reply by paradajz** [reply]
Sorry - haven't noticed HMI file when I extracted it. Just checked it. The design is fine. Still, what exactly is the issue? I don't understand.

**Reply by js-midi** [reply]
Only the first button has an on-off element applied and the rest of the on-off elements do not work.
However, MIDI signals work correctly.
Only the screen coordinates are causing the problem.

**Reply by paradajz** [reply]
Aha. Well you need to do some matching of triggers... How do you want to control the icon states on display - locally, via presses on the touchscreen, or via MIDI input?

**Reply by js-midi** [reply]
The button index value is recognized correctly and the MIDI signal works well, but
The web configurator recognizes only the first button on the screen page and the rest are unresponsive.

**Reply by paradajz** [reply]
Please read my question again.

**Reply by js-midi** [reply]
Ah, now I know what the question is.
The current testing state is local.
It's the same whether it's local or MIDI.
I want to change the on/off color of the buttons on the touch screen by receiving MIDI signal feedback.
Everything is going smoothly.
But only the first button changes, the rest don't change
This happens equally on all boards.

**Reply by js-midi** [reply]
In the touch screen tab of the opendeck configurator screen, the button coordinate value is applied only to button index 0, and all other buttons are ignored.
What's the problem?

**Comment by paradajz** [comment]
What happens when you go to the configurator, LED section, select LED 9(first touchscreen LED) and then color test it? Does the indicator change state? Same question for the other touchscreen LEDs.

**Reply by js-midi** [reply]
The first button works normally in color test.
The rest do not respond to the color test.

**Reply by js-midi** [reply]
https://youtu.be/ZU7jrBPqaw0

**Reply by paradajz** [reply]
I've found the underlying issue and it's now fixed on master branch.  Do you know how to compile firmware? If not I'll attach the binaries you need here.

**Reply by js-midi** [reply]
thank you so much.
It is not a compilable environment.
I want to get the binary file from here.

**Reply by paradajz** [reply]
For which boards?

**Reply by js-midi** [reply]
stm32f411 black pill

**Reply by paradajz** [reply]
It's attached. Just note that since this is directly from master branch, it's a development firmware and it will also wipe out your current configuration.

[blackpill411.zip](https://github.com/shanteacontrols/OpenDeck/files/7149275/blackpill411.zip)

**Reply by js-midi** [reply]
It goes very well.
Thank you again
